tests/DashboardControllerTest.php

```diff
@@ -0,0 +1,177 @@
+<?php
+
+namespace Cronbeat\Tests;
+
+use PHPUnit\Framework\Assert;
+use Cronbeat\Controllers\DashboardController;
+use Cronbeat\Database;
+use Cronbeat\RedirectException;
+use Cronbeat\Views\DashboardView;
+use Cronbeat\Views\MonitorFormView;
+
+class DashboardControllerTest extends DatabaseTestCase {
+    private ?DashboardController $controller = null;
+    private int $userId;
+    private string $username = 'testuser';
+    private string $passwordHash;
+
+    protected function setUp(): void {
+        parent::setUp();
+
+        $this->passwordHash = hash('sha256', 'password');
+        $this->getDatabase()->createUser($this->username, $this->passwordHash);
+        $this->userId = $this->getDatabase()->validateUser($this->username, $this->passwordHash);
+
+        $_SESSION = [];
+        $_SESSION['user_id'] = $this->userId;
+
+        $this->controller = new DashboardController($this->getDatabase());
+    }
+
+    protected function tearDown(): void {
+        $_SESSION = [];
+        parent::tearDown();
+    }
+
+    public function testDoRoutingRedirectsToLoginWhenNotAuthenticated(): void {
+        // Given
+        $_SESSION = []; // Clear session to simulate unauthenticated user
+
+        // Set up exception expectation
+        $this->expectException(RedirectException::class);
+        
+        // When
+        $this->controller->doRouting();
+    }
+    
+    public function testDoRoutingRedirectsToLoginWithCorrectHeaders(): void {
+        // Given
+        $_SESSION = []; // Clear session to simulate unauthenticated user
+        
+        // When
+        try {
+            $this->controller->doRouting();
+        } catch (RedirectException $e) {
+            // Then - Verify the exception contains the correct headers
+            $headers = $e->getHeaders();
+            $this->assertArrayHasKey('Location', $headers);
+            $this->assertEquals('/login', $headers['Location']);
+            return;
+        }
```

Comment:
@jetbrains-junie do not use try-catch but `expectException`. Check also other tests